src/Npgsql/TypeHandlers/CompositeHandlers/CompositeHandler.cs

```diff
@@ -90,25 +97,89 @@ public override int ValidateAndGetLength(T value, ref NpgsqlLengthCache? lengthC
             lengthCache.Set(0);
 
             // number of fields + (type oid + field length) * member count
-            var length = sizeof(int) + sizeof(int) * 2 * _memberHandlers.Length;
-            foreach (var member in _memberHandlers)
+            var memberHandlers = _memberHandlers!;
+            var length = sizeof(int) + sizeof(int) * 2 * memberHandlers.Length;
+            foreach (var member in memberHandlers)
                 length += member.ValidateAndGetLength(value, ref lengthCache);
 
             return lengthCache.Lengths[position] = length;
         }
 
-        CompositeMemberHandler<T>[] CreateMemberHandlers() => CreateMemberHandlers((PostgresCompositeType)PostgresType, _typeMapper, _nameTranslator);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void Initialize()
+        {
+            if (_memberHandlers is null)
+                InitializeCore();
+        }
 
-        static CompositeMemberHandler<T>[] CreateMemberHandlers(PostgresCompositeType pgType, ConnectorTypeMapper typeMapper, INpgsqlNameTranslator nameTranslator)
+        void InitializeCore()
```

Comment:
Move this to a local function inside Initialize? I think inlining should still work correctly.